### PR TITLE
mlos_bench config json schema validation - storage

### PR DESCRIPTION
**Pull request by bpkroth**
Extends #340 and #346 to storage configs.
See Also: #331

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -2,11 +2,14 @@
...
 # Specify a score threshold to be exceeded before program exits with error.
-fail-under=9.7
+fail-under=9.8
 
 # Make sure public methods are documented.
 # See Also: https://github.com/PyCQA/pydocstyle/issues/309#issuecomment-1426642147
-fail-on=C0116
+# Also fail on unused imports.
+fail-on=
+    C0116,
```
Comment (motus) [inline .pylintrc:11]: maybe write explicitly `missing-function-docstring` instead of cryptic C0116?

File: setup.cfg [hunk 2]
```diff
@@ -5,12 +5,11 @@ universal = 1
 [pycodestyle]
 count = True
 # E124: Closing bracket does not match indentation of opening bracket's line
-# E128: Continuation line under-indented for visual indent
 # E261: At least two spaces before inline comment
 # E502: The backslash is redundant between brackets
 # W503: Line break occurred before a binary operator
 # W504: Line break occurred after a binary operator
-ignore = E124,E128,E261,E502,W503,W504
+ignore = E124,E261,E502,W503,W504
```
Comment (motus) [inline setup.cfg:12]: I would allow failing on at least E124, E261, E502

**Review by motus — COMMENTED** [review]
LGTM! Let's merge in the first PR and I'll do the final review

File: .pylintrc (hunk 1 above)
Comment (bpkroth) [reply .pylintrc]: These were just to try and get my local checks failing in the same way as you were seeing with flake8 but without making too many unrelated changes.  Let's take that up in #358

File: setup.cfg (hunk 2 above)
Comment (bpkroth) [reply setup.cfg]: These were just to try and get my local checks failing in the same way as you were seeing with flake8 but without making too many unrelated changes. Let's take that up in https://github.com/microsoft/MLOS/pull/358

**Comment by bpkroth** [comment]
> LGTM! Let's merge in the first PR and I'll do the final review

Previous PR is merged.  Please have another look and approve when you get a sec.  Mostly just a little more test code consolidation.  Thanks!

**Review by motus — APPROVED** [review]

**Review by motus — APPROVED** [review]

**Pull request merged**